### DISCUSSION TITLE
Using select menus instead of index based selection

### DISCUSSION THREAD

**Discussion by bishoy-at-pieces** [pieces-app/cli-agent#140] [Ideas]
Currently we use indexes in the many commands eg. open,conversation
we do `open 1` to open the first asset for example 
or conversation 1 to get the first conversation, What do you think about using select menus instead

Small demo video:

https://github.com/pieces-app/cli-agent/assets/167128903/a2748328-1a78-4c25-8ef0-8eb5caa48d8a

**Comment by bishoy-at-pieces** [comment]
https://bishoyatpieces.pieces.cloud/?p=c787439ee9
Here is the code snippet for this basic select menus 

first, before we dig deeper
which commands do we need this to be in?
which commands will be deprecated?
what should be the command name after this eg: the list assets should be renamed to something else since it will also open after the listing

cc: @mark-at-pieces @mason-at-pieces @shivay-at-pieces 

Related issue: #133

**Reply by sambhavnoobcoder** [reply]
okay , addressing the points you mentioned , here are my opinions on those :

Which commands do we need this to be in?

We need to implement this new functionality in the commands pieces explore assets , pieces explore apps and pieces explore models

Which commands will be deprecated?

I think that ,No commands needs be deprecated. We can look at this like adding new functionality rather than replacing existing commands.
The current list commands can remain unchanged  , still supporting pieces list assets , pieces list apps and pieces list models commands.

What should be the command name after this?

I think we can name the new command `explore` instead of `list` for the interactive selection feature.
in this way, `pieces explore assets` will become `pieces list assets`  interactive version.
Similarly, we'll have pieces explore apps and pieces explore models.

Thus In this approach , The list commands remain as they are, providing simple listing functionality without interaction while The new explore commands provide both listing and interactive selection . We're not renaming existing commands, but rather introducing new ones with enhanced functionality. This structure allows us to Maintain backward compatibility with existing list commands while also clearly differentiating between simple listing (list) and interactive exploration (explore).

this Is what I have thought so far . did I miss anything out here ?

**Reply by bishoy-at-pieces** [reply]
So what will happen when you select an asset?
It should open it right? if so, the open command should be deprecated

**Reply by sambhavnoobcoder** [reply]
yeah , thats right , but in the case where we deprecate and remove the open completely , doesn't that make the list command a bit redundant ? if the select or explore command ( however we decide to name it ) does the features of list and more , would we want to replace the list command in its entirety , instead of keeping both options to list and the new select command open ?

**Reply by mason-at-pieces** [reply]
@Bishoy-at-pieces @sambhavnoobcoder Is there any reason why we would still want the original index implementation? So we'd simply just keep all commands the same with using `list` and show the selection instead.

**Reply by sambhavnoobcoder** [reply]
yeah , I thought it would be a bit redundant as well , but I was just confirming if we had any plans for future uses of exclusively the original index implementation or perhaps the open command since we are deprecating both , so the select would then essentially replace both .

**Reply by bishoy-at-pieces** [reply]
We can add a deprecated warning for this release that the open command will be deprecated, 
You can begin if you like, it should list the assets and open them
